subprojects/docs/src/docs/userguide/authoringMaintainableBuildScripts.adoc

```diff
@@ -42,17 +42,34 @@ $ gradle tasks
 
 > Task :tasks
 
-...
-
 Documentation tasks
 -------------------
 generateDocs - Generates the HTML documentation for this project.
-
-...
 ----
 
-[[sec:avoid_use_of_imperative_logic]]
-=== Avoid the use of imperative logic
+[[sec:minimize_logic_executed_configuration_phase]]
+=== Minimize logic executed during the configuration phase
```

Comment:
Would you say this belongs in the script maintainability doc? It seems covered pretty well in the Performance guide, which is sort of a part of the best practices docs.

Reply:
I think this topic is so essential that I wouldn't mind having duplicated coverage in different guides with a slightly different angle.

> Would you say this belongs in the script maintainability doc?

It is somewhat about where logic should live _within_ the script. Especially beginners make this mistake all the time. I don't think these users would necessarily look at the performance guide to figure out the issue as they haven't even understood the build lifecycle. _But_ I'd argue that they'd rather look at a best practices chapter.

I'd say the coverage in the performance guide is a bit abstract and doesn't show a concrete code example: https://guides.gradle.org/performance/#configuration. I don't feel that it is very approachable for beginners.